src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,325 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "my-service"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/my-service"
+                  rewrite = "/"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+
+
+##### Services mapped with preserved path
+
+The `preservePath` query parameter on a service declaration indicated that the context root path should be preserved when passing through ConductR's proxy. Given the service URI `http://my-service?preservePath`, the service is accessible from the outside world on HTTP port allocated by ConductR under `/my-sevice` context.
+
+This behaviour can now be declared with a request ACL.
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http://my-service?preservePath")))
+)
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "my-service",
+    RequestAcl(Http("^/my-service".r))
+  )
+)
+```
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        services      = ["http://my-service?preservePath"]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "my-service"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/my-service"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+#### Supplying customized HAProxy template configuration
+
+The following services definition will require a customized HAProxy configuration.
+
+The [[Dynamic proxy configuration|DynamicProxyConfiguration]] describes the process of generating a customized HAProxy configuration.
+
+Generating a customized HAProxy configuration will rely on the [[ifAcl|DynamicProxyConfiguration#ifAcl]] directive which accepts `system`, `systemVersion`, and `endpointLabel` as input argument.
+
+When declaring bundle settings via `build.sbt`, unless declared explicitly, the `system` and `systemVersion` settings may fallback to a default value as described by [sbt-conductr settings](https://github.com/typesafehub/sbt-conductr#bundle-settings).
+
+
+##### Customized port number
+
+A service may declare an endpoint exposed on a custom port number. Given the service URI `http://:6789/my-service`, the service is accessible from the outside world on HTTP port `6789` under `/my-service` context.
```

Comment:
@markusjura : explanation for customized port number - this is the most common case, i.e. `http://:6789/my-service`